url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveLexer.java

```diff
@@ -0,0 +1,15495 @@
+			}
+
+			switch (alt75) {
+				case 1 :
+					// HiveLexer.g:338:10: 'day'
+					{
+					match("day"); 
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:338:21: 'days'
+					{
+					match("days"); 
+
+					}
+					break;
+
+			}
+			state.type = _type;
+			state.channel = _channel;
+		}
+		finally {
+			// do for sure before leaving
+		}
+	}
+	// $ANTLR end "DAY"
+
+	// $ANTLR start "WEEK"
+	public final void mWEEK() throws RecognitionException {
+		try {
+			int _type = WEEK;
+			int _channel = DEFAULT_TOKEN_CHANNEL;
+			// HiveLexer.g:339:8: ( 'week' | 'weeks' | 'wks' ( DOT )? )
+			int alt77=3;
+			int LA77_0 = input.LA(1);
+			if ( (LA77_0=='w') ) {
+				int LA77_1 = input.LA(2);
+				if ( (LA77_1=='e') ) {
+					int LA77_2 = input.LA(3);
+					if ( (LA77_2=='e') ) {
+						int LA77_4 = input.LA(4);
+						if ( (LA77_4=='k') ) {
+							int LA77_5 = input.LA(5);
+							if ( (LA77_5=='s') ) {
+								alt77=2;
+							}
+
+							else {
+								alt77=1;
+							}
+
+						}
+
+						else {
+							int nvaeMark = input.mark();
+							try {
+								for (int nvaeConsume = 0; nvaeConsume < 4 - 1; nvaeConsume++) {
+									input.consume();
+								}
+								NoViableAltException nvae =
+									new NoViableAltException("", 77, 4, input);
+								throw nvae;
+							} finally {
+								input.rewind(nvaeMark);
+							}
+						}
+
+					}
+
+					else {
+						int nvaeMark = input.mark();
+						try {
+							for (int nvaeConsume = 0; nvaeConsume < 3 - 1; nvaeConsume++) {
+								input.consume();
+							}
+							NoViableAltException nvae =
+								new NoViableAltException("", 77, 2, input);
+							throw nvae;
+						} finally {
+							input.rewind(nvaeMark);
+						}
+					}
+
+				}
+				else if ( (LA77_1=='k') ) {
+					alt77=3;
+				}
+
+				else {
+					int nvaeMark = input.mark();
+					try {
+						input.consume();
+						NoViableAltException nvae =
+							new NoViableAltException("", 77, 1, input);
+						throw nvae;
+					} finally {
+						input.rewind(nvaeMark);
+					}
+				}
+
+			}
+
+			else {
+				NoViableAltException nvae =
+					new NoViableAltException("", 77, 0, input);
+				throw nvae;
+			}
+
+			switch (alt77) {
+				case 1 :
+					// HiveLexer.g:339:10: 'week'
+					{
+					match("week"); 
+
+					}
+					break;
+				case 2 :
+					// HiveLexer.g:339:21: 'weeks'
+					{
+					match("weeks"); 
+
+					}
+					break;
+				case 3 :
+					// HiveLexer.g:339:33: 'wks' ( DOT )?
+					{
+					match("wks"); 
+
+					// HiveLexer.g:339:39: ( DOT )?
+					int alt76=2;
+					int LA76_0 = input.LA(1);
+					if ( (LA76_0=='.') ) {
+						alt76=1;
+					}
+					switch (alt76) {
+						case 1 :
+							// HiveLexer.g:
+							{
+							if ( input.LA(1)=='.' ) {
+								input.consume();
+							}
+							else {
+								MismatchedSetException mse = new MismatchedSetException(null,input);
+								recover(mse);
+								throw mse;
+							}
+							}
+							break;
+
+					}
+
+					}
+					break;
+
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 9414:9460, 9673:9719.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.